packages/components/tests/integration/components/hds/code-block/index-test.js

```diff
@@ -0,0 +1,169 @@
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" id="test-code-block" />
+    `);
+    assert.dom('#test-code-block').doesNotHaveClass(/language-*/);
+  });
+
+  test('it uses the passed in language value for syntax highlighting', async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" @language="go" id="test-code-block" />
+    `);
+    assert.dom('#test-code-block').hasClass('language-go');
+  });
+
+  test('syntax highlighting fails gracefully if an invalid language is specified', async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" @language="foo" id="test-code-block" />
+    `);
+    assert.dom('#test-code-block').hasClass('language-foo');
+    assert.dom('#test-code-block .token').doesNotExist();
+  });
+
+  // hasCopyButton
+  test('it does not display a Copy button by default', async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" />
+    `);
+    assert.dom('.hds-code-block__copy-button').doesNotExist();
+  });
+
+  test(`it displays a Copy button if hasCopyButton is set to true`, async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" @hasCopyButton={{true}} />
+    `);
+    assert.dom('.hds-code-block__copy-button').exists();
+  });
+
+  // hasLineNumbers
+  test('it displays line numbers by default', async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" id="test-code-block" />
+    `);
+    assert.dom('#test-code-block').hasClass('line-numbers');
+  });
+
+  test('it does not display line numbers if hasLineNumbers is set to false', async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" @hasLineNumbers={{false}} id="test-code-block" />
+    `);
+    assert.dom('#test-code-block').doesNotHaveClass('line-numbers');
+  });
+
+  // hasLineWrapping
+  test('it does not wrap code line by default', async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" id="test-code-block" />
+    `);
+    assert
+      .dom('#test-code-block')
+      .doesNotHaveClass('hds-code-block--has-line-wrapping');
+  });
+
+  test('it wraps code lines if hasLineWrapping is set to true', async function (assert) {
+    await render(hbs`
+      <Hds::CodeBlock @value="console.log('Hello world');" @hasLineWrapping={{true}} id="test-code-block" />
+    `);
+    assert
+      .dom('#test-code-block')
+      .hasClass('hds-code-block--has-line-wrapping');
+  });
+
+  // highlightLines
+  test('it highlights the passed in individual line numbers', async function (assert) {
+    await render(hbs`
+    <Hds::CodeBlock
+      id="test-code-block-highlight"
+      @highlightLines={{"1"}}
+      @value="import Application from '@ember/application';
```

Comment:
since we are highlighting only the first line, maybe we could use a simpler code example here (the `console.log('hello world')` for example)

Reply:
Updated in [6d132d4](https://github.com/hashicorp/design-system/pull/1687/commits/6d132d4d13c138fac12293ea1b6b0ed947e87d11)